openedx/core/djangoapps/plugins/middleware.py

```diff
@@ -0,0 +1,19 @@
+from django.utils.deprecation import MiddlewareMixin
+
+
+class ViewNameAndSlotMiddleware(MiddlewareMixin):
+    """
+    Django middleware object to inject view name into request context
+    """
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        """
+        Injects the view name value into the request context
+        """
+        request.view_name = view_func.__name__
+        # For class-based views the view function will have a `view_class` attribute
+        # and we can get the slot_namespace from that
+        view = getattr(view_func, 'view_class', view_func)
+        request.slot_namespace = view.__qualname__
```

Comment:
I'm wondering if there should be no default slot namespace, so only specifically-namespaced views can be plugged into. Otherwise, the name of the view itself becomes part of the API contract you shouldn't break across releases.

Reply:
+1. If the view isn't marked as a "slot"ed view, just skip this step.

Reply:
Done